examples/structured_data/tabtransformer.py

```diff
@@ -0,0 +1,537 @@
+for training or evaluation.
+"""
+
+from tensorflow.keras.layers import StringLookup
+
+target_label_lookup = StringLookup(
+    vocabulary=TARGET_LABELS, mask_token=None, num_oov_indices=0
+)
+
+
+def prepare_example(features, target):
+    target_index = target_label_lookup(target)
+    weights = features.pop(WEIGHT_COLUMN_NAME)
+    return features, target_index, weights
+
+
+def get_dataset_from_csv(csv_file_path, batch_size=128, shuffle=False):
+    dataset = tf.data.experimental.make_csv_dataset(
+        csv_file_path,
+        batch_size=batch_size,
+        column_names=CSV_HEADER,
+        column_defaults=COLUMN_DEFAULTS,
+        label_name=TARGET_FEATURE_NAME,
+        num_epochs=1,
+        header=False,
+        na_value="?",
+        shuffle=shuffle,
+    ).map(prepare_example)
+    return dataset.cache()
+
+
+"""
+## Implement a training and evaluation procedure
+"""
+
+
+def run_experiment(model):
+
+    optimizer = tfa.optimizers.AdamW(
+        learning_rate=learning_rate, weight_decay=weight_decay
+    )
+
+    model.compile(
+        optimizer=optimizer,
+        loss=keras.losses.BinaryCrossentropy(),
+        metrics=[keras.metrics.BinaryAccuracy(name="accuracy")],
+    )
+
+    train_dataset = get_dataset_from_csv(train_data_file, batch_size, shuffle=True)
+    validation_dataset = get_dataset_from_csv(test_data_file, batch_size)
+
+    print("Start training the model...")
+    history = model.fit(
+        train_dataset, epochs=num_epochs, validation_data=validation_dataset
+    )
+    print("Model training finished")
+
+    _, accuracy = model.evaluate(validation_dataset, verbose=0)
+
+    print(f"Validation accuracy: {round(accuracy * 100, 2)}%")
+
+    return history
+
+
+"""
+## Create model inputs
+
+Now, define the inputs for the models as a dictionary, where the key is the feature name,
+and the value is a `keras.layers.Input` tensor with the corresponding feature shape
+and data type.
+"""
+
+
+def create_model_inputs():
+    inputs = {}
+    for feature_name in FEATURE_NAMES:
+        if feature_name in NUMERIC_FEATURE_NAMES:
+            inputs[feature_name] = layers.Input(
+                name=feature_name, shape=(), dtype=tf.float32
+            )
+        else:
+            inputs[feature_name] = layers.Input(
+                name=feature_name, shape=(), dtype=tf.string
+            )
+    return inputs
+
+
+"""
+## Encode features
+
+The `encode_inputs` method returns `encoded_categorical_feature_list` and `numerical_feature_list`.
+We encode the categorical features as embeddings, using a fixed `embedding_dims` for all the features,
+regardless their vocabulary sizes. This is required for the Transformer model.
+"""
+
+
+def encode_inputs(inputs, embedding_dims):
+
+    encoded_categorical_feature_list = []
+    numerical_feature_list = []
+
+    for feature_name in inputs:
+        if feature_name in CATEGORICAL_FEATURE_NAMES:
```

Comment:
Likewise, this function will be more reusable if it only relies on function arguments rather than global constants

Reply:
I fixed this in the hyperparameters. However, I would keep the dataset metadata as global constants.
LMK if you are OK with this.